lib/src/dbs/executor.rs

```diff
@@ -297,50 +296,38 @@ impl<'a> Executor<'a> {
 									None => stm.compute(&ctx, &opt, &self.txn(), None).await,
 								};
 								// Finalise transaction
-								match &res {
-									Ok(_) => match stm.writeable() {
-										true => self.commit(loc).await,
-										false => self.cancel(loc).await,
-									},
-									Err(_) => self.cancel(loc).await,
-								};
+								if res.is_ok() && stm.writeable() {
+									self.commit(loc).await;
+								} else {
+									self.cancel(loc).await;
+								}
 								// Return the result
 								res
```

Comment:
I noticed that the potential error created in `commit`/`cancel` wasn't checked/used. I preserved that behavior, but it seems like a transaction that fails to commit (due to contention in the KV store) might be erroneously reported as successful.